numpy/lib/arraypad.py

```diff
@@ -771,8 +764,7 @@ def _pad_ref(arr, pad_amt, method, axis=-1):
     ref_chunk2 = arr[ref_slice][rev_idx]
 
     if 'odd' in method:
-        edge_slice2 = tuple(slice(None) if i != axis else slice(x - 1, x)
-                            for (i, x) in enumerate(arr.shape))
+        edge_slice2 = _end_slice(arr.shape, 1, axis=axis)
```

Comment:
The alternative I was considering was `_axis_slice(arr.shape, lambda n: slice(n-1, n), axis=axis)`, which covers a few more uses, but I wasn't sure if that was clearer.

Reply:
Not clearer as written, but I think it would be clearer if one were to just pass in the required slice itself. This is possible, since, as I noted above, it is not necessary to know the actual dimension; here, one would just pass in `slice(-1, None)`.